server/src/test/java/org/elasticsearch/action/support/AutoCreateIndexTests.java

```diff
@@ -85,26 +85,36 @@ public void testAutoCreationDisabled() {
         AutoCreateIndex autoCreateIndex = newAutoCreateIndex(settings);
         String randomIndex = randomAlphaOfLengthBetween(1, 10);
         IndexNotFoundException e = expectThrows(IndexNotFoundException.class, () ->
-            autoCreateIndex.shouldAutoCreate(randomIndex, buildClusterState()));
+            autoCreateIndex.shouldAutoCreate(randomIndex, false, buildClusterState()));
         assertEquals("no such index [" + randomIndex + "] and [action.auto_create_index] is [false]", e.getMessage());
     }
 
     public void testAutoCreationEnabled() {
         Settings settings = Settings.builder().put(AutoCreateIndex.AUTO_CREATE_INDEX_SETTING.getKey(), true).build();
         AutoCreateIndex autoCreateIndex = newAutoCreateIndex(settings);
-        assertThat(autoCreateIndex.shouldAutoCreate(randomAlphaOfLengthBetween(1, 10), buildClusterState()), equalTo(true));
+        assertThat(autoCreateIndex.shouldAutoCreate(randomAlphaOfLengthBetween(1, 10), false, buildClusterState()), equalTo(true));
+    }
+
+    public void testAutoCreationEnabledWithNoAutoCreateFlagSet() {
```

Comment:
Where does the defaulting of request-level setting to index-level setting happen?

Reply:
I don't understand your question. 

What is the default of the setting? Or how are the settings adjusted for this test?

Reply:
What I meant is more: where is the code that checks the value of the index-level setting should the request-level setting be false.